test/functional/test-document-click.js

```diff
@@ -315,3 +320,68 @@ describe('test-document-click onDocumentElementClick_', () => {
     });
   });
 });
+
+describe('test-document-click onDocumentElementCapturedClick_', () => {
+
+  describe('usage of getElementByTagNameFromEventShadowDomPath_', () => {
+    it('should handle absence of path', () => {
+      expect(getElementByTagNameFromEventShadowDomPath_({}, 'A')).to.be.null;
+    });
+
+    it('should find first anchor in path', () => {
+      const evt = {path: [
+          {tagName: 'FOO'}, {tagName: 'A', item: 1}, {tagName: 'A', item: 2}]};
+      expect(getElementByTagNameFromEventShadowDomPath_(evt, 'A')).to.equal(
+          evt.path[1]);
+    });
+  });
+
+  describe('when including expansion url', () => {
+
+    it('should expand click_x/click_y', () => {
+      return createIframePromise().then(iframe => {
+        installUrlReplacementsService(iframe.win);
+        const replacements = urlReplacementsFor(iframe.win);
+        Math.random = function() { return 135; };
```

Comment:
Please use `sandbox.stub()` for this. Otherwise, `Math.random()` will be replaced for all subsequent tests.